lib/spack/spack/main.py

```diff
@@ -989,6 +989,11 @@ def _main(argv=None):
             env_format_error = e
 
     def add_environment_scope(priority):
+        if env_format_error:
```

Comment:
In my case, just above here, we raise `spack.environment.SpackEnvironmentConfigError` which has no `filename` attribute